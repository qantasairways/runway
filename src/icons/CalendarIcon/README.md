```js
<CalendarIcon height="50" width="50" />
```
