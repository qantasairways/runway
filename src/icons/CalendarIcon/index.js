import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

function CalendarIcon({ height, width, className }) {
  const pathId = generate();
  const maskId = generate();

  return (
    <svg
      width={width}
      height={height}
      className={className}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          id={pathId}
          d="M10 10h2v2h-2v-2zm3 0h2v2h-2v-2zm3 0h2v2h-2v-2zm-9 3h2v2H7v-2zm3 0h2v2h-2v-2zm3 0h2v2h-2v-2zm3 0h2v2h-2v-2zm-9 3h2v2H7v-2zm3 0h2v2h-2v-2zm3 0h2v2h-2v-2zm5-10V4.5a.5.5 0 0 0-.5-.5h-2a.5.5 0 0 0-.5.5V6h-5V4.5a.5.5 0 0 0-.5-.5h-2a.5.5 0 0 0-.5.5V6H5v14h15V6h-2zm-2-1h1v2h-1V5zM8 5h1v2H8V5zM6 9h13v10H6V9z"
        />
      </defs>
      <g fillRule="evenodd">
        <mask id={maskId}>
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <use xlinkHref={`#${pathId}`} />
        <g mask={`url(#${maskId})`}>
          <path d="M0 0h198v120H0z" />
        </g>
      </g>
    </svg>
  );
}

CalendarIcon.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  className: PropTypes.string
};

CalendarIcon.defaultProps = {
  height: '24',
  width: '24',
  className: ''
};

export default CalendarIcon;
