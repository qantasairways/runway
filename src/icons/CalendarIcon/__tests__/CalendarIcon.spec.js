import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import CalendarIcon from '..';

jest.mock('shortid');

describe('CalendarIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <CalendarIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<CalendarIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<CalendarIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<CalendarIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
