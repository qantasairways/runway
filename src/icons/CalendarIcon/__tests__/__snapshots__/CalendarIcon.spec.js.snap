// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="0 0 24 24"
  width="36px"
>
  <defs>
    <path
      d="M12 2c.368 0 .667.298.667.667v4.666a.667.667 0 0 1-1.334 0V2.667c0-.369.299-.667.667-.667zM6.667 2c.368 0 .666.298.666.667v4.666a.667.667 0 1 1-1.333 0V2.667C6 2.298 6.298 2 6.667 2zm10.666 0c.369 0 .667.298.667.667v4.666a.667.667 0 1 1-1.333 0V2.667c0-.369.298-.667.666-.667zm-2 5.333a2 2 0 0 0 3.334 1.5v9.834H5.333V8.833a2 2 0 0 0 3.334-1.5H10a2 2 0 1 0 4 0h1.333zM8.667 4H10v2H8.667V4zm10.666 0H22v18H2V4h2.667v2H4v14h16V6h-.667V4zM14 4h1.333v2H14V4z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
  >
    <mask
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <use
      fillRule="nonzero"
      xlinkHref="#1a2b3c"
    />
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M0 0h24v24H0z"
      />
    </g>
  </g>
</svg>
`;
